src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -17,12 +18,28 @@ namespace NuGet.VisualStudio
     [Export(typeof(IVsFrameworkCompatibility3))]
     public class VsFrameworkCompatibility : IVsFrameworkCompatibility2, IVsFrameworkCompatibility3
     {
+        private INuGetTelemetryProvider _telemetryProvider;
+
+        [ImportingConstructor]
+        public VsFrameworkCompatibility(INuGetTelemetryProvider telemetryProvider)
+        {
+            _telemetryProvider = telemetryProvider;
+        }
+
         public IEnumerable<FrameworkName> GetNetStandardFrameworks()
         {
-            return DefaultFrameworkNameProvider
-                .Instance
-                .GetNetStandardVersions()
-                .Select(framework => new FrameworkName(framework.DotNetFrameworkName));
+            try
+            {
```

Comment:
this time of delegation makes me cringe :D 
The problem is that the only other `alternative` I can suggest is to have a wrapper that uses reflection.

Reply:
Another alternative, which I implemented, is to call `.ToList()` at the end of each. This of course has multiple performance impacts. If the caller does `Whatever().First()`, then we'll now be evaluating every item in the result set, not just the first. Also, we'll always be doing extra allocations, but it's hard to imagine that these APIs are being used in such critical codepaths where that perf is important.